saltproc/abc.py

```diff
@@ -153,5 +172,75 @@ def update_depletable_materials(self, mats, dep_end_time):
 
         """
 
+    def read_plaintext_file(self, file_path):
+        """Reads the content of a plaintext file for use by other methods.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to file.
+
+        Returns
+        -------
+        file_lines : list of str
+            File lines.
+
+        """
+        file_lines = []
+        with open(file_path, 'r') as file:
+            file_lines = file.readlines()
+        return file_lines
+
+    @abstractmethod
+    def convert_nuclide_code_to_name(self, nuc_code):
+        """Converts depcode nuclide code to symbolic nuclide name.
+
+        Parameters
+        ----------
+        nuc_code : str
+            Nuclide code
+
+        Returns
+        -------
+        nuc_name : str
+            Symbolic nuclide name (`Am242m1`).
+
+        """
+
+    @abstractmethod
+    def _convert_name_to_nuccode(self, nucname):
+        """Converts depcode nuclide name to ZA nuclide code
+
+        Parameters
+        ----------
+        nucname : str
+            Nuclide namce
+
+        Returns
+        -------
+        nuc_name : str
+            ZA nuclide code
+
+        """
+
+    def preserve_simulation_files(self, step_idx):
+        """Move simulation input and output files
+        to unique a directory
 
+        Parameters
+        ----------
+        step_idx : int
 
+        """
+        step_results_dir = self.output_path / f'step_{step_idx}_data'
+        step_results_dir.mkdir(exist_ok=True)
+
+        file_path = lambda file : self.output_path / file
+        output_paths = list(map(file_path, self._OUTPUTFILE_NAMES))
+        input_paths = list(map(file_path, self._INPUTFILE_NAMES))
+        for file_path, fname in zip(output_paths, self._OUTPUTFILE_NAMES):
+            file_path.rename(step_results_dir / fname)
+        for file_path, fname in zip(input_paths, self._INPUTFILE_NAMES):
+            lines = self.read_plaintext_file(file_path)
+            with open(step_results_dir / fname, 'w') as out_file:
+                out_file.writelines(lines)
```

Comment:
I think a little white space can be a good thing. I also think this is pep8 compliant?
```suggestion
        for file_path, fname in zip(output_paths, self._OUTPUTFILE_NAMES):
            file_path.rename(step_results_dir / name)
            
        for file_path, fname in zip(input_paths, self._INPUTFILE_NAMES):
            lines = self.read_plaintext_file(file_path)
            with open(step_results_dir / fname, 'w') as out_file:
                out_file.writelines(lines)
```